.github/workflows/os_comp.yml

```diff
@@ -90,28 +90,49 @@ jobs:
         cfg:
           - CC: 'gcc'
             CXX: 'g++'
+            CC_LD: 'bfd'
+            CXX_LD: 'bfd'
+            AR: 'ar'
           - MESON_ARGS: '--unity=on -Ddefault_library=static'
             RUN_TESTS_ARGS: '--no-unittests'
             CC: 'gcc'
             CXX: 'g++'
+            CC_LD: 'bfd'
+            CXX_LD: 'bfd'
+            AR: 'ar'
           - MESON_ARGS: '-Ddefault_library=both'
             RUN_TESTS_ARGS: '--no-unittests'
             CC: 'gcc'
             CXX: 'g++'
+            CC_LD: 'bfd'
+            CXX_LD: 'bfd'
+            AR: 'ar'
           - CC: 'clang'
             CXX: 'clang++'
+            CC_LD: 'lld'
+            CXX_LD: 'lld'
+            AR: 'llvm-ar'
```

Comment:
Ubuntu's binutils is should be configured properly with plugin support, so this is incorrect. The tests should be testing that mixing clang and ld.bfd work here.

Reply:
> should be configured properly with plugin support, so this is incorrect

OK, but I don't know exactly how that's done.

Also, all of this goes back to what I put out for discussion a couple of time in this thread:  The blunt but simple new validation in this PR will simply complain if it finds clang/++ paired with gnu's bfd linker because we have at least one known case that's incompatible (i.e. enabling LTO in the compiler in conjunction with any other non-llvm-based linker/ar tool), so a choice is to either -
- use this simple but blunt toolchain family validation, point out mixing tools from different families has known incompatibilities (e.g. llvm/clang + LTO + [gnu/bfd ... or any non-llvm-based linker tool] ), in an unhelpfully opaque error message that might give users a little trouble, and ask the user to please build with tools all from the same family, which shouldn't be difficult to do ... although it's proving to not be trivial to iron out all these workflow job cases we have going on here that mix clang + bfd.
- or try to accommodate mixing tools across toolchains, while adding more special-cases to the validation and perhaps more tests, and perhaps risk this becoming increasingly complex, with more edge cases and more tests.
- or a new alternative would be to simply ditch any attempt at trying to help the user avoid incompatibilities, let the user mix whatever tool combinations they want and leave them to figure out that clang + LTO + ar/bfd is a bad combo.

I don't have a strong preference between the first 2 but it seems slightly more preferable (from a maintenance, testing, and avoiding surprise new incompatibilities point of view) to have users simply stick with tools from the same toolchain.

Are you firmly in the 2nd option or are you after something else?